corporate/lib/stripe.py

```diff
@@ -1264,13 +1271,139 @@ def get_initial_upgrade_context(
 
         return None, context
 
+    def downgrade_at_the_end_of_billing_cycle(self, plan: Optional[CustomerPlan] = None) -> None:
+        if plan is None:  # nocoverage
+            customer = self.get_customer()
```

Comment:
The story for no coverage here is:

Earlier, `test_downgrade()` in `test_stripe` resulted in calling `update_plan` and hence this function. Now, the `update_plan` codepath has `plan` argument already present, so this doesn't get coverage.

But, we mock this function in `test_support_view.test_downgrade_realm`, so this codenbock is necessary but we are mocking that, therefore no code coverage.

What should be do ideally here?

Reply:
I think it's probably fine to do without coverage on this detail. You could add a comment "TODO: This code path is used in production but mocked in current tests."